lib/kubernetes-deploy/kubernetes_resource/custom_resource_definition.rb

```diff
@@ -0,0 +1,40 @@
+# frozen_string_literal: true
+module KubernetesDeploy
+  class CustomResourceDefinition < KubernetesResource
+    TIMEOUT = 30.seconds
+    GLOBAL = true
+
+    def deploy_succeeded?
+      names_accepted_status == "True"
+    end
+
+    def deploy_failed?
+      names_accepted_status == "False"
+    end
+
+    def timeout_message
+      UNUSUAL_FAILURE_MESSAGE
```

Comment:
Let's be more specific about what happened here, e.g. "The names this CRD is attempting to register were neither accepted nor rejected in time"